fs-storage/src/file_storage.rs

```diff
@@ -293,40 +360,79 @@ mod tests {
     }
 
     #[test]
-    fn test_file_storage_is_storage_updated() {
+    fn test_file_metadata_timestamp_updated() {
         let temp_dir =
             TempDir::new("tmp").expect("Failed to create temporary directory");
         let storage_path = temp_dir.path().join("teststorage.txt");
 
         let mut file_storage =
-            FileStorage::new("TestStorage".to_string(), &storage_path);
+            FileStorage::new("TestStorage".to_string(), &storage_path).unwrap();
         file_storage.write_fs().unwrap();
-        assert_eq!(file_storage.needs_syncing().unwrap(), false);
         std::thread::sleep(std::time::Duration::from_secs(1));
+
         file_storage.set("key1".to_string(), "value1".to_string());
-        assert_eq!(file_storage.needs_syncing().unwrap(), true);
+        let before_write = fs::metadata(&storage_path)
+            .unwrap()
+            .modified()
+            .unwrap();
+        file_storage.write_fs().unwrap();
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
Since we now have a clever solution, can we remove all the delays from the tests?